File: 3394-minimum-array-end/minimum-array-end.cpp
class Solution {
public:
    long long minEnd(int n, int x) {
        vector<int>arr;
        for(int i=0;i<64;i++){
        if(x&(1ll<<i))
        continue;
        arr.push_back(i);
        }
        n--;
        int j=0;
        long long ans=x;
        //cout<<ans<<" "<<n;
        for(int i=0;i<64;i++)
        {
            if((n*1ll)&(1ll<<i))
            ans|=(1ll<<arr[j]);
            j++;
        }
        return ans;
    }
};

File: 3394-minimum-array-end/minimum-array-end.py
class Solution:
    def minEnd(self, n: int, x: int) -> int:
        arr=[]
        for i in range(64):
            if(x&(1<<i)):
                continue
            arr.append(i)
        n-=1
        j=0
        ans=x
        for i in range(64):
            if(n&(1<<i)):
                ans|=(1<<arr[j])
            j+=1
        return ans
        